### PR TITLE
Refactor Messages model and add chat images feature (Feature/chat pictures)

### PR DESCRIPTION
**Pull request by DeepGreen1**

### CODE REVIEW

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt [hunk 1]
```diff
@@ -32,14 +43,17 @@ class ChatActivity : AppCompatActivity() {
         initMessageList(model)
         initSendMessageButton(model)
         initSendLocationButton(model, locationManager, fusedLocationClient)
+        initSendImageButton(model)
+        //initSendImageCameraButton() // TODO: Implement this
```
Comment (pmcintyr) [inline app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:47]: Hello, good job on implementing the image messages! Is someone currently working on the initSendImageCameraButton or are you planning to do this in the future?

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt [hunk 2]
```diff
@@ -72,67 +86,86 @@ class ChatActivity : AppCompatActivity() {
...
+        viewModel.initAdapter()
+        messageList.adapter = viewModel.getAdapter()
+    }
+    */
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        val TAKE_PICTURE_REQUEST_CODE = 1
+        if (requestCode == TAKE_PICTURE_REQUEST_CODE && resultCode == RESULT_OK) {
```
Comment (pmcintyr) [inline app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:166]: I am struggling to understand the requestCode parameter, why is it used in onActivityResult?

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt [hunk 3]
```diff
@@ -72,67 +86,86 @@ class ChatActivity : AppCompatActivity() {
...
-//        val linearManager = LinearLayoutManager(this)
-//        linearManager.stackFromEnd = true
-//        recyclerView.layoutManager = linearManager
-//        recyclerView.setHasFixedSize(true)
-//
-//        viewModel.initAdapter()
-//        recyclerView.adapter = viewModel.getAdapter()
-//    }
+    // LOCALISATION METHODS (inspired by https://github.com/punit9l/)
+    // TODO: Refactor when the firebase helper is implemented
```
Comment (pmcintyr) [inline app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:109]: the firebase helper has been implemented last week, let's try to get this code uncommented & working relatively soon

**Review by pmcintyr — APPROVED** [review]

**Review by hamzahass — APPROVED** [review]
I don't see any issues with the code. However, it would be great to integrate image messages with firesbase, and also try to better the view in which images are portrayed in the chat.

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt (hunk 2 above)
Comment (DeepGreen1) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:166]: The requestCode parameter is used to differentiate the different results of activities that comes back to (in this case) ChatActivity.
In other words: when the user has taken a picture with the camera, the ChatActivity resumes.

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt (hunk 1 above)
Comment (DeepGreen1) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:47]: Thanks! Well it's not defined know who'll implement this feature. Hassan is currently working on the profile picture: this is basically the same thing i.e. selecting an image from galery and updating the UI. Thus, we'll reuse some of his work for this.

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt (hunk 3 above)
Comment (DeepGreen1) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:109]: Yes nice! That's work for next week (or next sprint). It's already planned.

**Pull request merged**